File: asml/LC102BinaryTreeLevelOrderTraversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> answer;
        vector<int> sublist;
        queue<TreeNode*> q;
        if (!root) return answer;
        q.push(root);
        while (!q.empty()) {
            int count = q.size();
            for (int i = 0; i < count; i++) {
                if (q.front()->left) q.push(q.front()->left);
                if (q.front()->right) q.push(q.front()->right);
                sublist.push_back(q.front()->val);
                q.pop();
            }
            answer.push_back(sublist);
            sublist.clear();
        }
        return answer;
    }
};

File: Linkedin/LC102BinaryTreeLevelOrderTraversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def levelOrder(self, root):
        if not root:
            return []
        queue = []
        answer = []
        next_level = []
        queue.append(root)
        temp = []
        while queue:
            while queue:
                node = queue.pop(0)
                temp.append(node.val)
                if node.left:
                    next_level.append(node.left)
                if node.right:
                    next_level.append(node.right)
            queue = next_level
            next_level = []
            answer.append(temp)
            temp = []
        return answer
